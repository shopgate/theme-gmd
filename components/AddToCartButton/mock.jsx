/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * Mocked AddToCartButton.
 * @type {MockedAddToCartButton}
 */
// eslint-disable-next-line react/prefer-stateless-function
export const MockedAddToCartButton = class extends Component {
  static propTypes = {
    handleAddToCart: PropTypes.func,
  };
  static defaultProps = {
    handleAddToCart: () => {},
  };
  /**
   * Renders mocked button.
   * @return {JSX}
   */
  render() {
    return (
      <button onClick={this.props.handleAddToCart} />
    );
  }
};
