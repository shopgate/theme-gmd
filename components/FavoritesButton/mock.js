/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

const mockedProduct1 = {
  productId: 'foo',
};

const currentProduct = {
  productId: 'bar',
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateNotOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId],
    },
  },
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
      [currentProduct.productId]: {
        productData: currentProduct,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId, currentProduct.productId],
    },
  },
};

/**
 * Mocked state without data.
 * @type {{product: {currentProduct: {}, favorites: {}}}}
 */
export const mockedStateEmpty = {
  product: {
    currentProduct: {},
    productsById: {},
  },
  favorites: {
    products: {
      ids: [],
    },
  },
};
