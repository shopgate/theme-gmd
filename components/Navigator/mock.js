import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const router = {
  routing: false,
  stack: [
    {
      id: '44a34767-2683-4414-9de4-fd736808fdab',
      params: {},
      pathname: '/',
      pattern: '/',
      query: {},
      state: {},
    },
  ],
};

export const defaultState = {
  navigator: {
    backgroundColor: 'red',
    enabled: true,
    filterOpen: false,
    filterAttributeOpen: false,
    loginOpen: false,
    searchActive: false,
    showCartIcon: true,
    showIconShadow: false,
    showProgressBar: true,
    showSearch: true,
    showTitle: true,
    textColor: 'white',
    viewTracking: false,
  },
  search: {
    suggestion: {},
  },
  router,
  view: {
    isLoading: false,
  },
  ui: {
    general: {
      title: null,
    },
  },
};

/**
 * Creates store.
 * @param {Object} props Props which would override state.navigator props.
 * @return {Object}
 */
export const getStore = props => configureStore([thunk])({
  ...defaultState,
  navigator: {
    ...defaultState.navigator,
    ...props,
  },
  search: {
    suggestions: {},
  },
});
