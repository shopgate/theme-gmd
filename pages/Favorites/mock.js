/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */
const foo = {
  productData: {
    name: 'Foo',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'foo',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

const bar = {
  productData: {
    name: 'Bar',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'bar',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

export const mockedEmptyState = {
  product: {
    productsById: {
      foo,
    },
  },
  favorites: {
    products: {
      ids: [],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: ['foo'],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedNotReadyState = {
  ...mockedEmptyState,
  favorites: {
    products: {},
    isFetching: false,
  },
};

export const mockedNextProps = {
  products: [
    foo.productData,
    bar.productData,
  ],
};

