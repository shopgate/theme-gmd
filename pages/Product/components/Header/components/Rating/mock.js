/*
 *  Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

/**
 * Mock for getElementById
 * @param {function} scrollSpy Element.scroll spy function.
 * @returns {function}
 */
const getElementById = scrollSpy => () => ({
  offsetTop: 100,
  closest() {
    return {
      scroll: scrollSpy,
    };
  },
});

export {
  getElementById,
};
