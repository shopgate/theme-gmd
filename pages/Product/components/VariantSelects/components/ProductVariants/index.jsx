/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import find from 'lodash/find';
import StepByStep from './helpers/StepByStep';

export default WrappedComponent => class extends Component {
  /**
   * These props are currently only used within componentWillReceiveProps and
   * shouldComponentUpdate. The linter complains about this.
   * @type {Object}
   */
  static propTypes = {
    getVariantsByProductId: PropTypes.func.isRequired,
    currentBaseProductId: PropTypes.string,
    getProductData: PropTypes.func,
    selection: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
      value: PropTypes.string,
      selected: PropTypes.bool.isRequired,
      disabled: PropTypes.bool.isRequired,
      values: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        selected: PropTypes.bool.isRequired,
        disabled: PropTypes.bool.isRequired,
        availability: PropTypes.shape({
          state: PropTypes.string.isRequired,
          text: PropTypes.string.isRequired,
        }),
      })).isRequired,
    })),
    setProductVariantId: PropTypes.func,
    variants: PropTypes.shape({
      products: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        hasOptions: PropTypes.bool.isRequired,
        characteristics: PropTypes.objectOf(PropTypes.string).isRequired,
      })).isRequired,
      characteristics: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        values: PropTypes.arrayOf(PropTypes.shape({
          id: PropTypes.string.isRequired,
          label: PropTypes.string.isRequired,
        })).isRequired,
      })).isRequired,
    }),
  };

  static defaultProps = {
    selection: null,
    variants: null,
    getProductData: () => {},
    setProductVariantId: () => {},
    currentBaseProductId: null,
  };

  /**
   * The component constructor
   * @param {Object} props The component props
   */
  constructor(props) {
    super(props);

    this.state = {
      selection: null,
    };

    this.selectHelper = null;
    this.productId = null;
  }

  /**
   * Fetch variants if available.
   */
  componentDidMount() {
    if (this.props.variants) {
      this.updateFromProps(this.props);
    }
  }

  /**
   * Initializes the selection helper as soon as product variants are available
   * @param {Object} nextProps Incoming props
   */
  componentWillReceiveProps(nextProps) {
    this.updateFromProps(nextProps);
  }

  /**
   * Makes sure to only update if really necessary
   * @param {Object} nextProps Incoming props
   * @param {Object} nextState Incoming state
   * @return {boolean}
   */
  shouldComponentUpdate(nextProps, nextState) {
    return this.selectHelper && Array.isArray(nextState.selection) && nextState.selection.length;
  }

  /**
   * Updates the component state from props.
   * @param {Object} [props] The props from which to update
   */
  updateFromProps(props = this.props) {
    const { variants, currentBaseProductId } = props;

    if (!this.selectHelper && variants) {
      /**
       * Initialize the helper. This only happens once per instance,
       * to avoid a reset of the component and it's children.
       * @type {StepByStep}
       */
      this.selectHelper = new StepByStep(variants).init();

      // Trigger the component state update
      this.updateSelectionState();
    } else {
      /**
       * Re-init for DOM cached pages.
       * We need this to sync the redux state with the component state again.
       */
      const productIdChangedFromNull = (
        this.props.currentBaseProductId === null && currentBaseProductId !== null
      );

      // If the productId was null, we know that we have to set the variantId in redux again.
      if (this.selectHelper && productIdChangedFromNull) {
        const variantId = this.productId;

        if (variantId) {
          const {
            variants: baseProductVariants = {},
          } = this.props.getVariantsByProductId(currentBaseProductId) || {};

          // Check if the variantId we want to set belongs to the currentProduct
          if (find(baseProductVariants.products, { id: variantId })) {
            // Trigger a setProductVariantId to bring the redux store back to the correct state.
            this.props.setProductVariantId(variantId);
          }
        }
      }
    }
  }

  /**
   * Creates new props for the wrapped component.
   * @return {Object} The props
   */
  createProps() {
    const { selection } = this.state;
    const props = { ...this.props };
    delete props.variants;

    return {
      ...props,
      selection,
      handleSelectionUpdate: (characteristicId, characteristicValueId) => {
        this.handleSelectionUpdate(characteristicId, characteristicValueId);
      },
    };
  }

  /**
   * Sets a characteristic value within a characteristic selection
   * @param {string} characteristicId A characteristic id
   * @param {string} characteristicValueId A characteristic value id
   * @return {ProductVariants}
   */
  handleSelectionUpdate(characteristicId, characteristicValueId) {
    if (this.selectHelper) {
      const updated = this.selectHelper.setCharacteristicValue(
        characteristicId,
        characteristicValueId
      );

      if (updated) {
        this.updateSelectionState();
      }
    }

    return this;
  }

  /**
   * Updates selection state
   * @return {ProductVariants}
   */
  updateSelectionState() {
    this.setState({
      selection: this.selectHelper.getSelection(),
    });

    const productId = this.selectHelper.getProductId();

    if (productId !== this.productId) {
      this.productId = productId;

      // Fetch data for the selected variant.
      this.props.getProductData(productId);
    }

    return this;
  }

  /**
   * Renders the underlying component
   * @returns {JSX}
   */
  render() {
    return (
      <WrappedComponent {...this.createProps()} />
    );
  }
};
