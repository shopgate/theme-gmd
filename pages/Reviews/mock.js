/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  mockedStateWithoutReview,
} from 'Components/Reviews/mock';

export const mockedState = {
  ...mockedStateWithoutReview,
  reviews: {
    reviewsByHash: {},
  },
  history: {
    state: {
      title: '',
      viewTop: true,
    },
  },
};
