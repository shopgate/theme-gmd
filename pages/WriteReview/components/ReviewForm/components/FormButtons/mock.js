/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked state with product only. Reviews not fetched.
 * @type {Object}
 */
export const mockedState = {
  history: {
    pathname: '/item/666f6f/write_review',
  },
  view: {
    isLoading: false,
  },
};
