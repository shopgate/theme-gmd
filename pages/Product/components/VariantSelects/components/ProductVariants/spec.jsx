/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import ProductVariants from './index';
import * as mockData from './mock';

/**
 * Wrapped mock component
 * @return {JSX}
 * @constructor
 */
const Mock = () => <div />;

Mock.propTypes = {
  handleSelectionUpdate: PropTypes.func.isRequired,
  selection: PropTypes.arrayOf(PropTypes.shape()),
  variants: PropTypes.shape(),
};

Mock.defaultProps = {
  selection: null,
  variants: null,
};

const WrappedMock = ProductVariants(Mock);
const store = createMockStore();

describe('<ProductVariants />', () => {
  it('shouldn\'t render without variants', () => {
    const wrapper = mount(<WrappedMock store={store} />);
    wrapper.setProps({ variants: null });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Mock).prop('selection')).toEqual(null);
    expect(wrapper.find(Mock).prop('variants')).toEqual(null);
  });

  it('should render with variants', () => {
    const wrapper = mount(<WrappedMock store={store} />);
    wrapper.setProps({ variants: mockData.variants });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Mock).prop('selection')).toEqual(mockData.initialSelection);
  });

  it('should update after selectionUpdate', () => {
    const wrapper = mount(<WrappedMock store={store} />);
    wrapper.setProps({ variants: mockData.variants });

    wrapper.find(Mock).prop('handleSelectionUpdate')('1', '1');
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Mock).prop('selection')).toEqual(mockData.selectionUpdate);
  });
});
