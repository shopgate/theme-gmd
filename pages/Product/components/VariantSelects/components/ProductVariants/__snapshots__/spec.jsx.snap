// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductVariants /> should render with variants 1`] = `
<_class
  currentBaseProductId={null}
  getProductData={[Function]}
  selection={null}
  setProductVariantId={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
  variants={
    Object {
      "characteristics": Array [
        Object {
          "id": "1",
          "label": "Color",
          "values": Array [
            Object {
              "id": "1",
              "label": "Black",
            },
            Object {
              "id": "2",
              "label": "Blue",
            },
          ],
        },
        Object {
          "id": "2",
          "label": "Size",
          "values": Array [
            Object {
              "id": "1",
              "label": "0",
            },
            Object {
              "id": "2",
              "label": "11",
            },
            Object {
              "id": "3",
              "label": "7",
            },
          ],
        },
      ],
      "products": Array [
        Object {
          "availability": Object {
            "state": "waning",
            "text": "Limited stock available",
          },
          "characteristics": Object {
            "1": "1",
            "2": "1",
          },
          "hasOptions": false,
          "id": "1013-1014",
        },
        Object {
          "availability": Object {
            "state": "alert",
            "text": "Out of stock",
          },
          "characteristics": Object {
            "1": "1",
            "2": "2",
          },
          "hasOptions": false,
          "id": "1013-1015",
        },
        Object {
          "availability": Object {
            "state": "ok",
            "text": "Available",
          },
          "characteristics": Object {
            "1": "1",
            "2": "3",
          },
          "hasOptions": false,
          "id": "1013-1016",
        },
        Object {
          "availability": Object {
            "state": "ok",
            "text": "Available",
          },
          "characteristics": Object {
            "1": "2",
            "2": "1",
          },
          "hasOptions": false,
          "id": "1013-1017",
        },
        Object {
          "availability": Object {
            "state": "ok",
            "text": "Available",
          },
          "characteristics": Object {
            "1": "2",
            "2": "2",
          },
          "hasOptions": false,
          "id": "1013-1018",
        },
        Object {
          "availability": Object {
            "state": "ok",
            "text": "Available",
          },
          "characteristics": Object {
            "1": "2",
            "2": "3",
          },
          "hasOptions": false,
          "id": "1013-1019",
        },
      ],
    }
  }
>
  <Mock
    currentBaseProductId={null}
    getProductData={[Function]}
    handleSelectionUpdate={[Function]}
    selection={
      Array [
        Object {
          "disabled": false,
          "id": "1",
          "label": "Color",
          "selected": false,
          "value": null,
          "values": Array [
            Object {
              "availability": null,
              "disabled": false,
              "id": "1",
              "label": "Black",
              "selected": false,
            },
            Object {
              "availability": null,
              "disabled": false,
              "id": "2",
              "label": "Blue",
              "selected": false,
            },
          ],
        },
        Object {
          "disabled": true,
          "id": "2",
          "label": "Size",
          "selected": false,
          "value": null,
          "values": Array [
            Object {
              "availability": null,
              "disabled": true,
              "id": "1",
              "label": "0",
              "selected": false,
            },
            Object {
              "availability": null,
              "disabled": true,
              "id": "2",
              "label": "11",
              "selected": false,
            },
            Object {
              "availability": null,
              "disabled": true,
              "id": "3",
              "label": "7",
              "selected": false,
            },
          ],
        },
      ]
    }
    setProductVariantId={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      }
    }
    variants={null}
  >
    <div />
  </Mock>
</_class>
`;

exports[`<ProductVariants /> should update after selectionUpdate 1`] = `
<_class
  currentBaseProductId={null}
  getProductData={[Function]}
  selection={null}
  setProductVariantId={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
  variants={
    Object {
      "characteristics": Array [
        Object {
          "id": "1",
          "label": "Color",
          "values": Array [
            Object {
              "id": "1",
              "label": "Black",
            },
            Object {
              "id": "2",
              "label": "Blue",
            },
          ],
        },
        Object {
          "id": "2",
          "label": "Size",
          "values": Array [
            Object {
              "id": "1",
              "label": "0",
            },
            Object {
              "id": "2",
              "label": "11",
            },
            Object {
              "id": "3",
              "label": "7",
            },
          ],
        },
      ],
      "products": Array [
        Object {
          "availability": Object {
            "state": "waning",
            "text": "Limited stock available",
          },
          "characteristics": Object {
            "1": "1",
            "2": "1",
          },
          "hasOptions": false,
          "id": "1013-1014",
        },
        Object {
          "availability": Object {
            "state": "alert",
            "text": "Out of stock",
          },
          "characteristics": Object {
            "1": "1",
            "2": "2",
          },
          "hasOptions": false,
          "id": "1013-1015",
        },
        Object {
          "availability": Object {
            "state": "ok",
            "text": "Available",
          },
          "characteristics": Object {
            "1": "1",
            "2": "3",
          },
          "hasOptions": false,
          "id": "1013-1016",
        },
        Object {
          "availability": Object {
            "state": "ok",
            "text": "Available",
          },
          "characteristics": Object {
            "1": "2",
            "2": "1",
          },
          "hasOptions": false,
          "id": "1013-1017",
        },
        Object {
          "availability": Object {
            "state": "ok",
            "text": "Available",
          },
          "characteristics": Object {
            "1": "2",
            "2": "2",
          },
          "hasOptions": false,
          "id": "1013-1018",
        },
        Object {
          "availability": Object {
            "state": "ok",
            "text": "Available",
          },
          "characteristics": Object {
            "1": "2",
            "2": "3",
          },
          "hasOptions": false,
          "id": "1013-1019",
        },
      ],
    }
  }
>
  <Mock
    currentBaseProductId={null}
    getProductData={[Function]}
    handleSelectionUpdate={[Function]}
    selection={
      Array [
        Object {
          "disabled": false,
          "id": "1",
          "label": "Color",
          "selected": true,
          "value": "1",
          "values": Array [
            Object {
              "availability": null,
              "disabled": false,
              "id": "1",
              "label": "Black",
              "selected": true,
            },
            Object {
              "availability": null,
              "disabled": false,
              "id": "2",
              "label": "Blue",
              "selected": false,
            },
          ],
        },
        Object {
          "disabled": false,
          "id": "2",
          "label": "Size",
          "selected": false,
          "value": null,
          "values": Array [
            Object {
              "availability": Object {
                "state": "waning",
                "text": "Limited stock available",
              },
              "disabled": false,
              "id": "1",
              "label": "0",
              "selected": false,
            },
            Object {
              "availability": Object {
                "state": "alert",
                "text": "Out of stock",
              },
              "disabled": false,
              "id": "2",
              "label": "11",
              "selected": false,
            },
            Object {
              "availability": null,
              "disabled": false,
              "id": "3",
              "label": "7",
              "selected": false,
            },
          ],
        },
      ]
    }
    setProductVariantId={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      }
    }
    variants={null}
  >
    <div />
  </Mock>
</_class>
`;

exports[`<ProductVariants /> shouldn't render without variants 1`] = `
<_class
  currentBaseProductId={null}
  getProductData={[Function]}
  selection={null}
  setProductVariantId={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
  variants={null}
>
  <Mock
    currentBaseProductId={null}
    getProductData={[Function]}
    handleSelectionUpdate={[Function]}
    selection={null}
    setProductVariantId={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      }
    }
    variants={null}
  >
    <div />
  </Mock>
</_class>
`;
